libnavigation-core/src/main/java/com/mapbox/navigation/core/navigator/NavigatorMapper.kt

```diff
@@ -127,6 +129,19 @@ private fun NavigationStatus.getRouteProgress(
                             route.directionsRoute.stepGeometryToPoints(legStep)
                         }
                         routeProgressCurrentState = routeState.convertState()
+
+                        stepGeometryIndex = shapeIndex - route.directionsRoute
+                            .stepsGeometryToPoints()[legIndex]
```

Comment:
This call would happen on the main thread and we don't truly have a guarantee that the cache is warmed up already (or that it wasn't cleared). Could look for a way to make this async?

Reply:
What if I only take the geometries of prev steps here? I mean they probably would be in the cache because we are not teleporting from step 0 to step 3.

Reply:
Implemeneted.